src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -1081,11 +1081,11 @@ public void getModifiedDocIds(DocIdPusher pusher)
                 log.log(Level.FINE, "latestTimestamp updated: {0}",
                     formatter.format(new Date(latestTimestamp.getTime())));
               }
-              if (lastDocIdTimestamp == null || ts.after(lastDocIdTimestamp)) {
-                lastDocIdTimestamp = ts;
-                docIdLastModifiedMap.put(id, lastDocIdTimestamp);
-                log.log(Level.FINE, "lastDocIdTimestamp updated: {0}",
-                    formatter.format(new Date(lastDocIdTimestamp.getTime())));
+              if (cachedLastModified == null) {
+                cachedLastModified = new Date(ts.getTime());
```

Comment:
We're constructing this same Date object twice here, maybe they could be shared, if it doesn't seem too artificial.